src/components/Payroll/RunPayrollFlow/RunPayrollFlow.stories.tsx

```diff
@@ -0,0 +1,352 @@
+      />
+      <DataView
+        label="Configuration"
+        columns={[
+          {
+            title: 'Employees',
+            render: () => <Text>John Smith</Text>,
+          },
+          {
+            title: 'Gross Pay',
+            render: () => <Text>$2,345.16</Text>,
+          },
+          {
+            title: 'Reimbursements',
+            render: () => <Text>$0.00</Text>,
+          },
+        ]}
+        data={[{}]}
+      />
+    </Flex>
+  )
+}
+
+// **** BLOCKS ****
+// **** BLOCKS ****
+// **** BLOCKS ****
+
+// TODO: Replace this hook with call to Speakeasy instead
+const useEditEmployeeApi = ({ employeeId }) => {
+  const mutate = async () => {}
+  return { mutate }
+}
+const PayrollEditEmployeeBlock = ({ employeeId, onEvent, onSaved }) => {
+  const { mutate } = useEditEmployeeApi({ employeeId })
+  const onDone = async () => {
+    await mutate()
+    onSaved()
+  }
+  return (
+    <BaseComponent onEvent={onEvent}>
+      <PayrollEditEmployee onDone={onDone} />
+    </BaseComponent>
+  )
+}
+
+// TODO: Replace this hook with call to Speakeasy instead
+const usePayrollApi = ({ payrollId }) => {
+  return {
+    data: {
+      employees: [{ employeeId: 'cdef' }],
+    },
+  }
+}
+// TODO: Replace this hook with call to Speakeasy instead
+const useCalculatePayrollApi = ({ payrollId }) => {
+  const mutate = async () => {}
+  return { mutate }
+}
+const PayrollConfigurationBlock = ({ onBack, onEvent, onCalculated, payrollId }) => {
+  const {
+    data: { employees },
+  } = usePayrollApi({ payrollId })
+  const { mutate } = useCalculatePayrollApi({ payrollId })
+  const [editedEmployeeId, setEditedEmployeeId] = useState(undefined)
+  const onCalculatePayroll = async () => {
+    await mutate({ payrollId })
+    onCalculated()
+  }
+  const onEdit = ({ employeeId }) => {
+    setEditedEmployeeId(employeeId)
+  }
+  const onSaved = () => {
+    setEditedEmployeeId(undefined)
+  }
+
+  const childComponent = editedEmployeeId ? (
+    <PayrollEditEmployeeBlock onEvent={onEvent} employeeId={editedEmployeeId} onSaved={onSaved} />
+  ) : (
+    <PayrollConfiguration
+      employees={employees}
+      onBack={onBack}
+      onCalculatePayroll={onCalculatePayroll}
+      onEdit={onEdit}
+    />
+  )
+
+  return <BaseComponent onEvent={onEvent}>{childComponent}</BaseComponent>
+}
+
+// TODO: Replace this hook with call to Speakeasy instead
+const useSubmitPayrollApi = ({ payrollId }) => {
+  const mutate = async () => {}
+  return { mutate }
+}
+const useListCompanyPayrollsApi = ({ companyId }) => {
+  return {
+    data: [{ payrollId: 'abcd' }],
+  }
+}
+export const DemoRunPayrollFlow = ({ companyId, onEvent }) => {
+  const [currentPayrollId, setCurrentPayrollId] = useState(undefined)
+  const [isCalculated, setIsCalculated] = useState(false)
+  const { data: payrolls } = useListCompanyPayrollsApi({ companyId })
+  const { mutate } = useSubmitPayrollApi({ payrollId: currentPayrollId })
+
+  const onEdit = () => {
```

Comment:
We can convert this to `useReducer` here instead of `useState`, I wasn't convinced initially that this was too much state wrangling but upon further reflection it might make comparison easier. Thoughts?